server/src/main/java/org/spine3/server/reflect/HandlerMethod.java

```diff
@@ -150,19 +165,18 @@ protected int getParamCount() {
      *                                   an {@link Error}. {@code Error} instances are propagated as-is.
      */
     protected <R> R invoke(Object target, Message message) throws InvocationTargetException {
-        checkNotNull(message);
+        return invoke(target, message, getDefaultContextInstance());
+    }
+
+    private C getDefaultContextInstance() {
```

Comment:
If we cannot avoid having this method, we need to cache its result per context class. We need to calculate it only once per class.